scripts/jenkins/ardana/ansible/roles/ardana_deploy/tasks/main.yml

```diff
@@ -97,9 +97,9 @@
   shell: grep  'public-KEY-API-extapi' /etc/hosts | awk {'print $1'}
   register: public_vip
 
-- name: Update /etc/hosts
-  lineinfile:
-    dest=/etc/hosts
-    line="{{ public_vip.stdout }}    {{ external_name }}"
-  become: yes
+- name: Update /etc/hosts on the deployer and controller nodes
+  shell: ansible -b -m lineinfile -a "dest=/etc/hosts line='{{ public_vip.stdout }}    {{ external_name }}'" \
+  ARD-SVC--first-member,*control-plane-1-controller:children,*control-plane-1-core:children
```

Comment:
You should use `resources` to cover all the non-deployer nodes, it is more generic. Also `OPS-LM--first-member` instead of `ARD-SVC--first-member`, for the same reason.

Reply:
If you want the non-deployer nodes then `resources:!OPS-LM--first-member` should give you the list of non-deployer nodes

Reply:
That of course assumes the deployer is the first node in the list of OPS-LM nodes if there is more than on node in the OPS-LM cluster ;-)

Reply:
I did not want to pick up all the non-deployer nodes like lmm, swift, dbmq, swift and neutron nodes (as in midscale model). I just wanted to pickup only deployer and controller/core nodes.